Data/ByteString/Short/Internal.hs

```diff
@@ -446,6 +563,838 @@ concat sbss =
+                           i = create nl $ \mba -> copyByteArray (asBA sbs) 0 mba 0 nl
+                       in Just (i, l')
+
+
+-- ---------------------------------------------------------------------
+-- Transformations
+
+-- | /O(n)/ 'map' @f xs@ is the ShortByteString obtained by applying @f@ to each
+-- element of @xs@.
+--
+-- @since 0.11.3.0
+map :: (Word8 -> Word8) -> ShortByteString -> ShortByteString
+map f = \sbs ->
+    let l = length sbs
+        ba = asBA sbs
+    in create l (\mba -> go ba mba 0 l)
+  where
+    go :: BA -> MBA s -> Int -> Int -> ST s ()
+    go !ba !mba !i !l
+      | i >= l = return ()
+      | otherwise = do
+          let w = indexWord8Array ba i
+          writeWord8Array mba i (f w)
+          go ba mba (i+1) l
+
+
+-- | /O(n)/ 'reverse' @xs@ efficiently returns the elements of @xs@ in reverse order.
+--
+-- @since 0.11.3.0
+reverse :: ShortByteString -> ShortByteString
+reverse = \sbs ->
+    let l = length sbs
+        ba = asBA sbs
+-- https://gitlab.haskell.org/ghc/ghc/-/issues/21015
+#if MIN_VERSION_base(4,12,0) && defined(SAFE_UNALIGNED)
+    in create l (\mba -> go ba mba l)
+  where
+    go :: forall s. BA -> MBA s -> Int -> ST s ()
+    go !ba !mba !l = case l `quotRem` 8 of
+      (0, r) -> void $ goWord8Chunk 0 r
+      (q, 0) -> goWord64Chunk 0 0 q
+      (q, r) -> do
+        i' <- goWord8Chunk 0 r
+        goWord64Chunk i' 0 q
+     where
+      -- Bodigrim: "For powers of 2 separate quot and rem are faster than quotRem."
+      quotRem :: Int -> Int -> (Int, Int)
+      quotRem x i = let q = x `quot` i
+                        r = x `rem` i
+                    in (q, r)
+
+      goWord64Chunk :: Int -> Int -> Int -> ST s ()
+      goWord64Chunk !off !i' !cl = loop i'
+       where
+        loop :: Int -> ST s ()
+        loop !i
+          | i >= cl = return ()
+          | otherwise = do
+              let w = indexWord64Array ba (off + (i * 8))
+              writeWord64Array mba (cl - 1 - i) (byteSwap64 w)
+              loop (i+1)
+
+      goWord8Chunk :: Int -> Int -> ST s Int
+      goWord8Chunk !i' !cl = loop i'
+       where
+        loop :: Int -> ST s Int
+        loop !i
+          | i >= cl = return i
+          | otherwise = do
+              let w = indexWord8Array ba i
+              writeWord8Array mba (l - 1 - i) w
+              loop (i+1)
+#else
+    in create l (\mba -> go ba mba 0 l)
+   where
+    go :: BA -> MBA s -> Int -> Int -> ST s ()
+    go !ba !mba !i !l
+      | i >= l = return ()
+      | otherwise = do
+          let w = indexWord8Array ba i
+          writeWord8Array mba (l - 1 - i) w
+          go ba mba (i+1) l
+#endif
+
+
+-- | /O(n)/ The 'intercalate' function takes a 'ShortByteString' and a list of
+-- 'ShortByteString's and concatenates the list after interspersing the first
+-- argument between each element of the list.
+--
+-- @since 0.11.3.0
+intercalate :: ShortByteString -> [ShortByteString] -> ShortByteString
+intercalate inc = \case
```

Comment:
What does `inc` stand for? How about `sep` for separator?!

Reply:
inclusor

Reply:
I've never heard of that word and can't find any relevant references. `inc` suggests `increment` to me. I suggest switching to `sep` or something neutral like `x`.

Reply:
> I've never heard of that word and can't find any relevant references. `inc` suggests `increment` to me. I suggest switching to `sep` or something neutral like `x`.

It's latin!

Reply:
I don't think we should expect people who read this code to know latin and to guess that `inc` stands for `inclusor`.